src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -60,7 +60,16 @@ public function registerClasses(Definition $prototype, $namespace, $resource, $e
         $interfaces = array();
         $singlyImplemented = array();
 
-        foreach ($classes as $class) {
+        foreach ($classes as $class => $errorMessage) {
+            if (null !== $errorMessage) {
```

Comment:
what about putting this "if" just after the existing `$this->setDefinition($class, unserialize($serializedPrototype));` line?
no need to explicitly create a new definition, the prototype-based one is better, isn't it?

Reply:
I agree about the prototype definition. About moving this block down: I'd like to avoid the `interface_exists()` call below, so the block should probably stay here.

Reply:
> I'd like to avoid the interface_exists() call below

is there any reason for that? I'm not sure this is a valid concern :)

Reply:
I expected the `interface_exists` call to trigger the autoloader again, but appearently it doesn't. I'm moving the block.

Reply:
The call uses `false` as second argument, precisely to avoid the autoloading call.

Reply:
Exactly. Block moved. ✅